### ISSUE TITLE
Track 404 errors

### ISSUE THREAD

**Issue by samwilson** [DaSchTour/matomo-mediawiki-extension#46]
The [Matomo docs suggest](https://matomo.org/faq/how-to/faq_60/) tracking 404 errors like this:

> Setup a custom 404 page: you can easily do this if you use a CMS (for example, in wordpress, all 404 pages are loading a custom 404 template). If you don’t use a Content Management System, you can usually specify a 404 custom page in your webserver configuration file.
> Add the Matomo tracking code in this custom 404 page. In the Javascript tracking code, add the following:
> `_paq.push(['setDocumentTitle',  '404/URL = ' +  encodeURIComponent(document.location.pathname+document.location.search) + '/From = ' + encodeURIComponent(document.referrer)]);`

Is this something that this extension could handle?

**Comment by samwilson** [comment]
There's some more information here, and an alternative way to track 404s: https://github.com/matomo-org/matomo/issues/6580

**Comment by Seb35** [comment]
Indeed, it would be something useful to understand what the visitors do not find – MediaWiki has already the special pages Wanted* ([WantedPages](https://www.mediawiki.org/wiki/Special:WantedPages), [WantedCategories](https://www.mediawiki.org/wiki/Special:WantedCategories), …) but it’s more from the point of view of the editors (what pages should we create?) instead of the visitors (what pages do we want to see? including pages not anticipated by the editors).

Given Matomo has no official method to support that (but it could appear in the future), perhaps the best way for now is to use one of the method described in the links in the parameter `$wgMatomoCustomJS` (which is written before the call to `trackPageView`):
* [modify the title](https://matomo.org/faq/how-to/faq_60/), and/or
* [add custom variables](https://www.webfactory.de/blog/piwik-track-missing-pages-404-errors-with-custom-variables-and-segmentation) and/or
* possibly [a custom dimension](https://matomo.org/guide/reporting-tools/custom-dimensions/)

And experiment what is the more convenient. When a method will be stable, it could be documented [on the mediawiki.org page](https://www.mediawiki.org/wiki/Extension:Matomo#Matomo_configuration_optimized_for_MediaWiki).

**Comment by Seb35** [comment]
> And experiment what is the more convenient. When a method will be stable, it could be documented [on the mediawiki.org page](https://www.mediawiki.org/wiki/Extension:Matomo#Matomo_configuration_optimized_for_MediaWiki).

Ah, I realise some support from this extension is necessary to conditionnally include this JS snippet only on the 404 page, or perhaps more generally add a JS variable with the HTTP status code to be retrieved with `mw.config.get( 'wgHTTPStatus' );`

EDIT: and I see your PR #47.

**Comment by samwilson** [comment]
Yeah it's odd that Matomo doesn't have a better way to track them, but changing the title seems to work okay. The adding of the slashes means that it displays it grouped in the same way it does other directories.

Does that PR look ok?

**Comment by samwilson** [comment]
@Seb35 Gentle reminder about this. Let me know if you want me to update the PR in any way.
